src/panels/JobMonitor/ExecutionUnit.js

```diff
@@ -1,8 +1,69 @@
 import React                from 'react';
 import CollapsibleWidget    from 'paraviewweb/src/React/Widgets/CollapsibleWidget';
+import LogFold              from './LogFold';
 import { formatTime }       from '../../utils/Format';
 import style                from 'HPCCloudStyle/JobMonitor.mcss';
 
+// takes array => jsx lines
+function logMapper(entry, index) {
+  let content = '';
+  let foldContent = null;
+  let msg = entry.msg;
+  if (msg !== null && typeof msg === 'object') {
+    msg = JSON.stringify(msg, null, 2);
+  }
+
+  if (entry.status) {
+    msg += ` [${entry.status}]`;
+  }
+
+  content += `[${formatTime(entry.created)}] ${entry.levelname}: ${msg}`;
+
+  if (entry.exc_info) {
+    foldContent = entry.exc_info[2].join('');
+    foldContent += `${entry.exc_info[0]}: ${entry.exc_info[1]}`;
+  }
+
+  if (entry.data && Object.keys(entry.data).length > 0) {
+    foldContent = `${JSON.stringify(entry.data, null, 2)}`;
+  }
+
+  if (foldContent !== null) {
+    return (<LogFold key={`${entry.created}_${index}`}
+      header={content} content={foldContent} />);
+  }
+
+  return (<p key={`${entry.created}_${index}`} className={style.logEntry}>
+    {content}
+  </p>);
+}
+
+// const logReducer = (prevVal, entry, index) => {
+//   let content = prevVal;
+//   let msg = entry.msg;
+//   if (msg !== null && typeof msg === 'object') {
+//     msg = JSON.stringify(msg, null, 2);
+//   }
+
+//   if (entry.status) {
+//     msg += ` [${entry.status}]`;
+//   }
+
+//   content += `[${formatTime(entry.created)}] ${entry.levelname}: ${msg}\n`;
+
+//   if (entry.exc_info) {
+//     content += entry.exc_info[2].join('');
+//     content += `${entry.exc_info[0]}: ${entry.exc_info[1]}`;
+//   }
+
+//   if (entry.data && Object.keys(entry.data).length > 0) {
+//     const data = JSON.stringify(entry.data, null, 2);
+//     content += `${data}\n`;
+//   }
+
+//   return content;
+// };
+
```

Comment:
yes, that's the old logger function. >_<

On Thu, May 12, 2016 at 11:25 AM, Chris Harris <notifications@github.com>
wrote:

> In src/panels/JobMonitor/ExecutionUnit.js
> <https://github.com/Kitware/HPCCloud/pull/390#discussion_r63061245>:
>
> > +
> > +//   content += `[${formatTime(entry.created)}] ${entry.levelname}: ${msg}\n`;
> > +
> > +//   if (entry.exc_info) {
> > +//     content += entry.exc_info[2].join('');
> > +//     content += `${entry.exc_info[0]}: ${entry.exc_info[1]}`;
> > +//   }
> > +
> > +//   if (entry.data && Object.keys(entry.data).length > 0) {
> > +//     const data = JSON.stringify(entry.data, null, 2);
> > +//     content += `${data}\n`;
> > +//   }
> > +
> > +//   return content;
> > +// };
> > +
>
> Should we delete this?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Kitware/HPCCloud/pull/390/files/72298e34f7098667f892caa12e94c6c1fb754ef0#r63061245>
>



-- 
- Tristan Wright

*R&D Engineer*
*Kitware Inc.*